lib/granite/rubocop/cop/performer_in_initializer.rb

```diff
@@ -0,0 +1,198 @@
+          end
+        end
+
+        def reject_performer_pair(corrector, params, performer_param_pair)
+          performer_param_i = performer_param_index(params)
+          return unless performer_param_i
+
+          if params.children.size == 1
+            remove_full_parameters(corrector, params, performer_param_pair) && return
+          end
+
+          if performer_param_i < params.children.size - 1
+            remove_all_from_performer_param_until_next_param(performer_param_i, params, corrector)
+          else
+            remove_performer_param_only(performer_param_i, params, corrector)
+          end
+        end
+
+        def performer_param_index(params)
+          params.children.find_index { |pair| pair.children.first.children.first == :performer }
+        end
+
+        def remove_all_from_performer_param_until_next_param(performer_param_i, params, corrector)
+          performer_param = params.children[performer_param_i]
+          next_param = params.children[performer_param_i + 1]
+
+          begin_pos = performer_param.loc.expression.begin_pos
+          end_pos = next_param.loc.expression.begin_pos
+
+          remove_in_range(corrector, begin_pos, end_pos)
+        end
+
+        def remove_performer_param_only(performer_param_i, params, corrector)
+          performer_param = params.children[performer_param_i]
+          prev_param = params.children[performer_param_i - 1]
+
+          begin_pos = prev_param.loc.expression.end_pos
+          end_pos = performer_param.loc.expression.end_pos
+
+          remove_in_range(corrector, begin_pos, end_pos)
+        end
+
+        def remove_in_range(corrector, begin_pos, end_pos)
+          buffer = processed_source.buffer
+          range = Parser::Source::Range.new(buffer, begin_pos, end_pos)
+          corrector.remove(range)
+        end
+
+        def params_and_performer_pair_expression(node)
+          has_performer_as_expression(node) do |performer_param_pair|
+            params = performer_param_pair.parent.parent
+            return params, performer_param_pair
+          end
+        end
+
+        def params_and_performer_pair_role(node)
+          has_performer_as_role(node) do |performer_param_pair|
+            params = performer_param_pair.parent
+            return params, performer_param_pair
+          end
+        end
+
+        def remove_full_parameters(corrector, params, performer_param_pair)
+          if perform_merge_on_param_hash?(params)
+            corrector.replace params.loc.expression, '{}'
+          else
+            corrector.remove params.loc.expression
+            prev_sibling = previous_sibling(params)
+            corrector.remove range_between(prev_sibling, performer_param_pair) if prev_sibling.respond_to? :loc
+          end
+        end
+
+        def perform_merge_on_param_hash?(params)
+          params.parent.type == :send && params.parent.children.include?(:merge)
```

Comment:
Also `reverse_merge`/`merge`. And probably `{**a, **b}`.
And if there is a local variable somewhere in the scope that is initialized with a hash that includes `performer: ...`, and is later merged with `new` arguments.

Will it work if a BA.new is detected without preceding call to `as`?

Reply:
Can we assume that this cases will be caught by the deprecation warning? I think this cop should help with some auto-correcting, but maybe no need to handle all the quirky edge-cases.